From eee599899566ba654b052f2c4e242aed870cd41c Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 3 Jun 2014 06:01:51 +1000
Subject: [PATCH 1/2] uxa: fix pending

---
 src/uxa/intel.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/uxa/intel.h b/src/uxa/intel.h
index 6ac770e..f9dbd96 100644
--- a/src/uxa/intel.h
+++ b/src/uxa/intel.h
@@ -334,7 +334,7 @@ typedef struct intel_screen_private {
 	Bool has_kernel_flush;
 	Bool needs_flush;
 
-	struct _DRI2FrameEvent *pending_flip[2];
+	struct _DRI2FrameEvent *pending_flip[3];
 
 	/* Broken-out options. */
 	OptionInfoPtr Options;
-- 
1.7.1

