diff -up intel-gpu-tools-20130625/lib/drmtest.c.jx intel-gpu-tools-20130625/lib/drmtest.c
--- intel-gpu-tools-20130625/lib/drmtest.c.jx	2013-06-25 11:39:53.000000000 -0400
+++ intel-gpu-tools-20130625/lib/drmtest.c	2013-06-25 14:44:16.915552824 -0400
@@ -980,9 +980,7 @@ static struct format_desc_struct {
 	int depth;
 } format_desc[] = {
 	DF(RGB565,	RGB16_565,	16, 16),
-	DF(RGB888,	INVALID,	24, 24),
 	DF(XRGB8888,	RGB24,		32, 24),
-	DF(XRGB2101010,	RGB30,		32, 30),
 	DF(ARGB8888,	ARGB32,		32, 32),
 };
 #undef DF
