diff -up xf86-video-intel-2.99.911/src/intel_module.c.jx xf86-video-intel-2.99.911/src/intel_module.c
--- xf86-video-intel-2.99.911/src/intel_module.c.jx	2014-03-11 08:05:50.000000000 -0400
+++ xf86-video-intel-2.99.911/src/intel_module.c	2014-05-07 14:57:04.790360101 -0400
@@ -447,11 +447,15 @@ _xf86findDriver(const char *ident, XF86C
 	return NULL;
 }
 
-static enum accel_method { SNA, UXA, GLAMOR } get_accel_method(void)
+static enum accel_method { SNA, UXA, GLAMOR } get_accel_method(int gen)
 {
 	enum accel_method accel_method = DEFAULT_ACCEL_METHOD;
 	XF86ConfDevicePtr dev;
 
+	/* RHEL6: keep UXA default below Broadwell */
+	if (gen < 0100)
+	    accel_method = UXA;
+
 	if (hosted())
 		return SNA;
 
@@ -481,6 +485,7 @@ intel_scrn_create(DriverPtr		driver,
 		  unsigned		flags)
 {
 	ScrnInfoPtr scrn;
+	int gen;
 
 	if (match_data == 0) {
 		int devid = intel_entity_get_devid(entity_num), i;
@@ -497,6 +502,7 @@ intel_scrn_create(DriverPtr		driver,
 		if (match_data == 0)
 			return FALSE;
 	}
+	gen = ((struct intel_device_info *)match_data)->gen;
 
 	scrn = xf86AllocateScreen(driver, flags);
 	if (scrn == NULL)
@@ -513,12 +519,12 @@ intel_scrn_create(DriverPtr		driver,
 	xf86AddEntityToScreen(scrn, entity_num);
 
 #if UMS
-	if ((unsigned)((struct intel_device_info *)match_data)->gen < 020)
+	if (gen < 020)
 		return lg_i810_init(scrn);
 #endif
 
 #if KMS
-	switch (get_accel_method()) {
+	switch (get_accel_method(gen)) {
 #if USE_SNA
 	case SNA: return sna_init_scrn(scrn, entity_num);
 #endif
